lib/generate-doc.js

```diff
@@ -82,8 +82,25 @@ function iterateStack (path, routeLayer, layer, cb) {
   layer.route.stack.forEach((l) => iterateStack(path + layer.route.path, layer, l, cb))
 }
 
+function processComplexMatch (thing, keys) {
```

Comment:
I think we need some comments here. This is looking for the key and then replacing it with the key template syntax?  There are a bunch of replaces there I am not sure I am following. An approach I like to use is to pull all the regex's out and name them variable that explain what they match. Ideally outside of the function scope so they are recreated each call for no reason.

Reply:
Yeah, that was my thinking (add it to code comments). I think I followed most of this but it takes a bit and comments and/or var names would go along way to making this maintainable longer term and for folks who might not be as deep into it as we are.

Reply:
Mind adding one over the middle block just to cover all of them?